LeaderboardBackend/Models/Entities/BaseEntity.cs

```diff
@@ -0,0 +1,38 @@
+using System.ComponentModel.DataAnnotations;
+using NodaTime;
+
+/// <summary>
+///     Base class that provides common fields to all Entities.
+/// </summary>
+public class BaseEntity
+{
+	/// <summary>
+	///     The time this entity was created.
+	/// </summary>
+	[Required]
+	public Instant CreatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+	/// <summary>
+	///     The time this entity was updated.
+	/// </summary>
+	[Required]
+	public Instant UpdatedAt { get; set; } = SystemClock.Instance.GetCurrentInstant();
+
+	/// <summary>
+	///     The time this entity was deleted.
+	/// </summary>
+	public Instant? DeletedAt { get; set; }
+
+	public override bool Equals(object? obj)
+	{
+		return obj is BaseEntity entity
+			&& CreatedAt == entity.CreatedAt
+			&& UpdatedAt == entity.UpdatedAt
+			&& DeletedAt == entity.DeletedAt;
+	}
+
+	public override int GetHashCode()
```

Comment:
This doesn't seem right at all. Shouldn't the hash code be based on ID?

Reply:
Real talk I don't know what I'm doing half the time with overriding these methods. So it should just be based on ID usually?